### PR TITLE
Deprecate use of added mass via hydrodynamics

### PR DESCRIPTION
**Pull request by arjo129**
# 🦟 Bug fix

Addresses #1923

## Summary
Sneaking this in before feature freeze for ionic. I believe we should deprecate this option as it has instabilities (see #1923 for more context). The `fluid_added_mass` approach is superior. We should update our tutorials as well.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/systems/hydrodynamics/Hydrodynamics.cc [hunk 1]
```diff
@@ -340,7 +340,11 @@ void Hydrodynamics::Configure(
       << "\thttps://github.com/gazebosim/gz-sim/pull/1888" << std::endl;
   }
 
+
   // Added mass according to Fossen's equations (p 37)
+  // Note: Adding added mass here is deprecated and will be removed in
+  // Gazebo J as this formulation has instabilities.
+  bool added_mass_specified = false;
```
Comment (caguero) [inline src/systems/hydrodynamics/Hydrodynamics.cc:347]: Nit: Use camelCase instead.

File: src/systems/hydrodynamics/Hydrodynamics.cc [hunk 2]
```diff
@@ -350,12 +354,23 @@ void Hydrodynamics::Configure(
       prefix += "Dot";
       prefix += snameConventionVel[j];
       this->dataPtr->Ma(i, j) = SdfParamDouble(_sdf, prefix, 0);
+      if (std::abs(this->dataPtr->Ma(i, j)) > 1e-6)
+      {
+        added_mass_specified = true;
```
Comment (caguero) [inline src/systems/hydrodynamics/Hydrodynamics.cc:359]: Considering this is in a `for` loop, I think you want to change it to:

`addedMassSpecified = addedMassSpecified && (std::abs(this->dataPtr->Ma(i, j)) > 1e-6)`

You can then remove the `if` statement.

**Review by caguero — CHANGES_REQUESTED** [review]
There are some lines longer than 80 chars. See https://github.com/gazebosim/gz-sim/actions/runs/10098157483/job/27924575770?pr=2493

File: src/systems/hydrodynamics/Hydrodynamics.cc [hunk 3]
```diff
@@ -350,12 +354,23 @@ void Hydrodynamics::Configure(
       prefix += "Dot";
       prefix += snameConventionVel[j];
       this->dataPtr->Ma(i, j) = SdfParamDouble(_sdf, prefix, 0);
+      addedMassSpecified &= (std::abs(this->dataPtr->Ma(i, j)) > 1e-6);
```
Comment (caguero) [inline src/systems/hydrodynamics/Hydrodynamics.cc:357]: Strictly speaking `&=` is equal to `<exp> & <exp>`, the bitwise operator, not the logical one. I wouldn't use it here, just to be safe.

Comment (arjo129) [reply src/systems/hydrodynamics/Hydrodynamics.cc]: Thanks for spotting this! Fixed da9dc71bf.

**Review by iche033 — APPROVED** [review]

**Comment by azeey** [comment]
There's a warning on the Noble CI, but I don't think it's related to this. I'll go ahead and merge.

**Pull request merged**